src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java

```diff
@@ -434,4 +444,75 @@ else if(isFixed(loansPolicy)) {
       return Optional.empty();
     }
   }
+
+  public HttpResult<Loan> recall(Loan loan) {
+    final JsonObject recalls = representation
+        .getJsonObject("requestManagement", new JsonObject())
+        .getJsonObject("recalls", new JsonObject());
+
+    final HttpResult<DateTime> minimumDueDateResult =
+        getDueDate("minimumGuaranteedLoanPeriod", recalls,
+            loan.getLoanDate(), null);
+
+    // Use a java.time Clock, which allows unit tests to use a fixed clock
+    final DateTime systemDate = new DateTime(
+        ZonedDateTime.now(
+            ClockManager.getClockManager().getClock()).toInstant().toEpochMilli(),
+        DateTimeZone.UTC);
+
+    final HttpResult<DateTime> recallDueDateResult =
+        getDueDate("recallReturnInterval", recalls, systemDate, systemDate);
+
+    final List<ValidationError> errors = new ArrayList<>();
+
+    errors.addAll(combineValidationErrors(recallDueDateResult));
+    errors.addAll(combineValidationErrors(minimumDueDateResult));
+
+    if (errors.isEmpty()) {
+      final DateTime mgd = minimumDueDateResult.value();
+      final DateTime rd = recallDueDateResult.value();
+
+      final DateTime newDueDate;
+      if (mgd == null || rd.isAfter(mgd)) {
+        newDueDate = rd;
+      } else {
+        newDueDate = mgd;
+      }
+
+      loan.changeDueDate(newDueDate);
+
+      return succeeded(loan);
+    } else {
+      return failed(new ValidationErrorFailure(errors));
+    }
+  }
+
+  private HttpResult<DateTime> getDueDate(String key,
+      JsonObject representation,
+      DateTime initialDateTime,
+      DateTime defaultDateTime) {
+    final HttpResult<DateTime> result;
+
+    if (representation.containsKey(key)) {
+      result = getPeriod(representation, key).addTo(initialDateTime,
+          () -> failure(errorForPolicy(String.format("the \"%s\" in the loan policy is not recognized", key))), 
+          interval -> failure(errorForPolicy(String.format("the interval \"%s\" in \"%s\" is not recognized", interval, key))),
```

Comment:
I wonder if we want to move these errors to a method for creating a period that returns a result, and so can be used anywhere a period is created?

Reply:
Can you elaborate a bit more on this? For some reason I am not groking what you mean. Do people still use "grok" these days? I hope so. Stranger in a Strange Land is one of my favorite books. I am rambling... 🤐